### PR TITLE
add loop over exception

### PR DESCRIPTION
**Pull request by hchoura**
@lchanmann Please can you review.

### CODE REVIEW

File: src/foam/core/CompoundException.java [hunk 1]
```diff
@@ -32,7 +32,10 @@ public void add(Throwable t) {
 
   public void maybeThrow() {
     if ( exceptions_.size() != 0 ) {
-      throw this;
+      for ( int i = 0 ; i < exceptions_.size() ; i++ ) {
```
Comment (jlhughes) [inline src/foam/core/CompoundException.java:35]: @hchoura no.
Do this output when AppConfig.MODE is TEST.
after this block 
'throw this' as before.

File: src/foam/core/CompoundException.java [hunk 2]
```diff
@@ -32,7 +32,10 @@ public void add(Throwable t) {
...
+        System.out.println( "This is an error =======> " + ( (Throwable) exceptions_.get(i) ) );
+        throw (RuntimeException) exceptions_.get(i);
```
Comment (jlhughes) [inline src/foam/core/CompoundException.java:37]: this makes no sense.

**Review by jlhughes — CHANGES_REQUESTED** [review]

File: src/foam/core/CompoundException.java [hunk 3]
```diff
@@ -32,7 +32,10 @@ public void add(Throwable t) {
 
   public void maybeThrow() {
     if ( exceptions_.size() != 0 ) {
-      throw this;
```
Comment (lchanmann) [inline src/foam/core/CompoundException.java:35]: `throw this;` makes more sense than looping the exception list then throws the first one.

**Comment by jlhughes** [comment]
superseded by another PR

**Pull request closed without merging**